relays/client-crab/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+use std::time::Duration;
+
+use codec::Encode;
+use frame_support::weights::{IdentityFee, Weight};
+use sp_core::{storage::StorageKey, Pair};
+use sp_runtime::{generic::SignedPayload, traits::IdentifyAccount};
+
+use bp_messages::MessageNonce;
+use relay_substrate_client::{
+	Chain, ChainBase, ChainWithBalances, ChainWithMessages, SignParam, TransactionSignScheme,
+	UnsignedTransaction,
+};
+
+pub mod runtime;
+
+/// Crab header id.
+pub type HeaderId = relay_utils::HeaderId<bp_crab::Hash, bp_crab::BlockNumber>;
+
+/// Rialto header type used in headers sync.
+pub type SyncHeader = relay_substrate_client::SyncHeader<bp_crab::Header>;
+
+/// Millau chain definition.
+#[derive(Debug, Clone, Copy)]
+pub struct CrabChain;
+
+impl ChainBase for CrabChain {
+	type BlockNumber = bp_crab::BlockNumber;
+	type Hash = bp_crab::Hash;
+	type Hasher = bp_crab::Hashing;
+	type Header = bp_crab::Header;
+
+	type AccountId = bp_crab::AccountId;
+	type Balance = bp_crab::Balance;
+	type Index = bp_crab::Nonce;
+	type Signature = bp_crab::Signature;
+
+	fn max_extrinsic_size() -> u32 {
+		bp_crab::Crab::max_extrinsic_size()
+	}
+
+	fn max_extrinsic_weight() -> Weight {
+		bp_crab::Crab::max_extrinsic_weight()
+	}
+}
+
+impl Chain for CrabChain {
+	const NAME: &'static str = "Crab";
+	const TOKEN_ID: Option<&'static str> = Some("polkadot");
```

Comment:
polkadot ?

Reply:
Don't care, this field is used to update the currency exchange rate, we use free-market, so do not use it.